node/core/approval-voting/src/import.rs

```diff
@@ -461,11 +442,25 @@ pub(crate) async fn handle_new_head<Context, B: Backend>(
 			force_approve,
 		} = imported_block_info;
 
-		let session_info = state
-			.session_window
-			.as_ref()
-			.and_then(|s| s.session_info(session_index))
-			.expect("imported_block_info requires session info to be available; qed");
+		let session_info = match get_session_info(
+			session_info_provider,
+			ctx.sender(),
+			head,
+			session_index,
+		)
+		.await
+		{
+			Some(session_info) => session_info,
+			None => {
+				gum::error!(
```

Comment:
again `get_session_info` is already logging. If we want more precise logs, e.g. saying that it was for a new head: That is fine as well, but then we should reallly get rid of `get_session_info` as it does nothing useful then.

Reply:
Fixed for all `get_session_info()` calls. Only the log line inside the function is kept.